### ISSUE TITLE
False positive when checking consistency of **kwargs and positional arguments

### ISSUE THREAD

**Issue by dejfh** [microsoft/pyright#7473]
Pyright detects an error (reportInconsistentOverload), I expect no error.

Code example:

```python
from typing import overload

@overload
def fun(d: dict[str, float], /) -> None: ...

@overload
def fun(**kwargs: float) -> None: ...

def fun(d: dict[str, float] | None = None, /, **kwargs: float) -> None:
    pass
```

```
[...] - error: Overloaded implementation is not consistent with signature of overload 2
    Type "(d: dict[str, float] | None = None, /, **kwargs: float) -> None" cannot be assigned to type "(**kwargs: float) -> None"
      Parameter 1: type "float" cannot be assigned to type "dict[str, float] | None"
        Type "float" cannot be assigned to type "dict[str, float] | None"
          "float" is incompatible with "dict[str, float]"
          "float" is incompatible with "None" (reportInconsistentOverload)
```

Occurs in VS Code with Pylance (v2024.2.3) as well as with pyright command-line tool (1.1.353).

**Comment by dejfh** [comment]
Probably the same issue, but simpler:

```python
from typing import Protocol

def fun(d: int = 5, /, **kwargs: float) -> None:
    pass

class TheProtocol(Protocol):
    def __call__(self, **kwargs: float) -> None: ...

the_var: TheProtocol = fun
```

```
[...] - error: Expression of type "(d: int = 5, /, **kwargs: float) -> None" cannot be assigned to declared type "TheProtocol"
    Type "(d: int = 5, /, **kwargs: float) -> None" cannot be assigned to type "(**kwargs: float) -> None"
      Parameter 1: type "float" cannot be assigned to type "int"
        "float" is incompatible with "int" (reportAssignmentType)
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.355, which I just published.
